kasa/protocol.py

```diff
@@ -29,7 +30,24 @@
 _NO_RETRY_ERRORS = {errno.EHOSTDOWN, errno.EHOSTUNREACH, errno.ECONNREFUSED}
 
 
-class TPLinkSmartHomeProtocol:
+class TPLinkProtocol(ABC):
+    """Base class for all TP-Link Smart Home communication."""
+
+    def __init__(self, host: str, *, port: Optional[int] = None) -> None:
+        """Create a protocol object."""
+        self.host = host
+        self.port = port
```

Comment:
I think it makes sense to define the `credentials` also here as kwarg-only, right?

Reply:
Done